keps/sig-apps/2232-jobs-stopped/README.md

```diff
@@ -0,0 +1,522 @@
+definition](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning)
+or by redefining what graduation means.
+
+In general we try to use the same stages (alpha, beta, GA), regardless of how the
+functionality is accessed.
+
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+Below are some examples to consider, in addition to the aforementioned [maturity levels][maturity-levels].
+
+#### Alpha -> Beta Graduation
+
+- Gather feedback from developers and surveys
+- Complete features A, B, C
+- Tests are in Testgrid and linked in KEP
+
+#### Beta -> GA Graduation
+
+- N examples of real-world usage
+- N installs
+- More rigorous forms of testing—e.g., downgrade tests and scalability tests
+- Allowing time for feedback
+
+**Note:** Generally we also wait at least two releases between beta and
+GA/stable, because there's no opportunity for user feedback, or even bug reports,
+in back-to-back releases.
+
+#### Removing a Deprecated Flag
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+
+**For non-optional features moving to GA, the graduation criteria must include 
+[conformance tests].**
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+-->
+
+### Upgrade / Downgrade Strategy
+
+Upgrading from 1.20 and below will not change the behaviour of how Jobs work.
+
+To make use of this feature, the `StoppedJob` feature gate must be explicitly
+enabled on the API server and the controller manager and the `stopped` field
+must be explicitly set in the Job spec.
+
+### Version Skew Strategy
+
+The change is entirely limited to the control plane. Version skew across
+control plane / kubelet does not change anything.
+
+In HA clusters, version skew across different replicas in the control plane
+should also work seamlessly because only one controller manager will be active
+at any given time.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [x] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: `StoppedJob`
+    - Components depending on the feature gate:
+      - kube-apiserver
+      - kube-controller-manager
+
+* **Does enabling the feature change any default behavior?** No. Unless the
+  `stopped` field is explicitly set, default behaviour will remain the same.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?** Yes. Turning off the feature gate will disable the
```

Comment:
What will happen with jobs that has the Stopped field set?

Reply:
The job controller will ignore the stopped field if the feature gate isn't enabled. So if you restart the controller manager and there were stopped jobs, they will be assumed to be active (even though the stopped field is set in etcd; it will just be ignored).

I'll clarify this in the KEP.